ubuntu22.04/nvidia-driver

```diff
@@ -480,6 +482,22 @@ _shutdown() {
 }
 
 _resolve_kernel_type() {
+
+  # For backwards compatibility with older GPU Operator versions where KERNEL_MODULE_TYPE is not set,
+  # honor the deprecated OPEN_KERNEL_MODULES enabled field
+  if [ -z "${KERNEL_MODULE_TYPE}" ]; then
+   if [ ${OPEN_KERNEL_MODULES_ENABLED} == "true" ]; then
+     echo "Honoring deprecated field OPEN_KERNEL_MODULES_ENABLED=${OPEN_KERNEL_MODULES_ENABLED}"
+     KERNEL_TYPE=kernel-open
+     return 0
+   fi
+
+   # If both KERNEL_MODULE_TYPE and OPEN_KERNEL_MODLES_ENABLED are unset, default
+   # to the open kernel modules on newer driver branches
+   [[ "${DRIVER_BRANCH}" -gt "560" ]] && KERNEL_TYPE=kernel-open || KERNEL_TYPE=kernel
```

Comment:
What about the edge cases ? Drivers that only support one type of kernel modules?